core/src/sql/v2/index.rs

```diff
@@ -52,6 +52,9 @@ pub struct SearchParams {
 #[revisioned(revision = 2)]
 pub struct MTreeParams {
 	pub dimension: u16,
+	#[revision(start = 1, end = 2, convert_fn = "convert_old_distance")]
+	pub _distance: Distance1, // TODO remove once 1.0 && 1.1 are EOL
```

Comment:
I'm curious about this. Do we have a test to make sure this works with revision 1 objects since it renames the field name?

Reply:
Good point! In fact, Bincode is designed for speed and efficiency. So the fields are serialized in the order they are defined in the struct, whatever the name. Consequently, when deserializing, Bincode expects the fields in the binary data to be in the same order to correctly map them back to the struct fields.
So the name of the field does not matter.

The upgrade test includes a test that checks that the distance is the right one. 1.1.x is using revision 1, and 1.2.x is using revision 2.

Reply:
Yes bincode is fine because of the index-based keys, it's our manual serialiser implementations I was worried about. I don't think upgrading triggers those manual serialiser implementations since we will be deserialising objects already created by the previous version.

Anyway I think my concern is outside the scope of this PR because, now that I think about it, it's downgrading I'm worried about, not upgrading.

Reply:
In the manual serialiser, as it using the names, `_distance` and `distance` are considered synonyms, and deserialized by the same `Distance` serialiser.

https://github.com/surrealdb/surrealdb/blob/5f2d3b0a495936c5581f72365c706cb276241495/core/src/sql/v1/value/serde/ser/index/mtreeparams.rs#L69